sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -235,9 +239,16 @@ def is_in_notebook(self):
   @property
   def inspector(self):
     """Gets the singleton InteractiveEnvironmentInspector to retrieve
-    information consumable by other applications."""
+    information consumable by other applications such as a notebook
+    extension."""
     return self._inspector
 
+  @property
+  def inspector_with_synthetic(self):
+    """Gets the singleton InteractiveEnvironmentInspector with additional
+    synthetic variables generated by Interactive Beam. Internally used."""
+    return self._inspector_with_synthetic
+
```

Comment:
Why have a separate inspector? Specifically, why separate the notion of having an inspector with and without synthetic PCollections?

Reply:
The synthetic PCollections are the ones created when caching unbounded source outputs.

The inspector was originally created to list all the PCollections and pipelines defined in the notebook and then display them in Jupyter. Since the synthetic ones are not defined by the user, the should be hidden from them in that use case.

However, internally, when looking for the cacheables, the synthetic ones are needed to correctly create WriteCache transforms. Thus, we need a separate inspector. Or at least the inspector should produce 2 views of PCollections: one with the synthetics and one without.